torch/csrc/autograd/profiler.cpp

```diff
@@ -88,39 +95,59 @@ void popRange() {
   }
 }
 
-void enableProfiler(ProfilerState new_state) {
+void enableProfiler(ProfilerConfig config) {
+  ProfilerState new_state = config.state;
   AT_ASSERT(new_state != ProfilerState::Disabled);
   if (new_state == ProfilerState::NVTX && !cuda_stubs->enabled())
-    throw std::runtime_error("Can't use NVTX profiler - PyTorch was compiled without CUDA");
+    throw std::runtime_error(
+        "Can't use NVTX profiler - PyTorch was compiled without CUDA");
   if (state != ProfilerState::Disabled && new_state != state) {
-      throw std::runtime_error("can't change kind of profiling (e.g. NVTX to CPU) while profiler is running");
+    throw std::runtime_error(
+        "can't change kind of profiling (e.g. NVTX to CPU) while profiler is running");
   }
 
-  pushCallback([](const RecordFunction& fn) {
-    auto* msg = (fn.seqNr() >= 0) ? ", seq = " : "";
-    pushRangeImpl(fn.name(), msg, fn.seqNr());
-  },
-  [](const RecordFunction& /* unused */) {
-    popRange();
-  });
+  pushCallback(
+      [config](const RecordFunction& fn) {
```

Comment:
profiling with shape recording enabled will have much more overhead than without. is it worth leaving a note in the python docstring saying so?

Reply:
Good point, I will add it